androidScanner/app/src/main/java/com/example/androidScanner/MainActivity.java

```diff
@@ -158,25 +146,31 @@ public void onScanResult(int callbackType, ScanResult result) {
         }
     };
 
-    public void startScanning() {
+    public void startScanning(View v) {
+        startForegroundService(v);
         results.setText("");
-
         try {
-            client = new MqttClient("tcp://iot.ubikampus.net", MqttClient.generateClientId(), new MemoryPersistence());
+            SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
+            String serverURL = sharedPreferences.getString(MQTT_SERVER,"tcp://iot.ubikampus.net");
```

Comment:
Could we reuse `MQTT_SERVER` constant here? So that iot.ubikampus.net address is only defined once.

Reply:
iot.ubikampus.net is now hardcoded